nodebuilder/config_test.go

```diff
@@ -33,3 +34,104 @@ func TestConfigWriteRead(t *testing.T) {
 		})
 	}
 }
+
+// TestUpdateConfig tests that updating an outdated config
+// using a new default config applies the correct values and
+// preserves old custom values.
+func TestUpdateConfig(t *testing.T) {
+	cfg := new(Config)
+	_, err := toml.Decode(outdatedConfig, cfg)
+	require.NoError(t, err)
+
+	newCfg := DefaultConfig(node.Light)
+	// ensure this config field is not filled in the outdated config
+	require.NotEqual(t, newCfg.Share.PeerManagerParams, cfg.Share.PeerManagerParams)
+
+	cfg, err = updateConfig(cfg, newCfg)
+	require.NoError(t, err)
+	// ensure this config field is now set after updating the config
+	require.Equal(t, newCfg.Share.PeerManagerParams, cfg.Share.PeerManagerParams)
+	// ensure old custom values were not changed
+	require.Equal(t, "thisshouldnthavechanged", cfg.State.KeyringAccName)
+	require.Equal(t, "7979", cfg.RPC.Port)
+	require.True(t, cfg.Gateway.Enabled)
+}
+
+// outdatedConfig is an outdated config from a light node
+var outdatedConfig = `
```

Comment:
Woow, do we have to change this string everytime we change the config?

Reply:
No, it's an `outdatedConfig` :) it's purposefully old.

Reply:
Ok, cool